src/pages/library/datasetBuilder/DatasetBuilderDetails.ts

```diff
@@ -97,7 +107,10 @@ export const DatasetBuilderDetails = ({ datasetId }: DatasetBuilderDetailsProps)
               ]),
             ]),
           ]),
-          h(TileDisplay, { title: 'EHR Domains', displayInformation: datasetDetails.state.domainOptions }),
+          h(TileDisplay, {
+            title: 'EHR Domains',
+            displayInformation: datasetDetails.state?.snapshotBuilderSettings?.domainOptions ?? [],
```

Comment:
Since it's inside a block where `datasetDetails.status === 'Ready'`, `datasetDetails.state` is known by typescript to be defined.
```suggestion
            displayInformation: datasetDetails.state?.snapshotBuilderSettings?.domainOptions ?? [],
```